atlp-rwanda/Dominators-Phantom-FE
# Ft create roles permissions

#### What does this PR do?
- Create Roles and Permissions
- Set Roles' permissions 
- CRUD operations related to Roles and Permissions
#### Description of Task to be completed?
- Have the route `localhost:port/roles` working
#### How should this be manually tested?
- run `git checkout ft-create-roles-permissions`
- then `git pull origin ft-create-roles-permissions`
- then `npm install`
- make sure to add `BACKEND_URL=local API server` to your `.env` file 
- run `npm run dev`
- then check the route for roles  `localhost:port/roles`
#### Any background context you want to provide?
- I used React Context API 
#### What are the relevant pivotal tracker stories?
The relevant pivotal tracker stories can be found [here](https://trello.com/c/T02wU18e/32-create-roles-set-permissions)
#### Screenshots
![localhost:port/roles page](https://user-images.githubusercontent.com/81187224/174974438-fbbb3132-36fc-41a7-bec8-71af5cdf832d.png)
![Adding a new Role](https://user-images.githubusercontent.com/81187224/174974819-3a179ab9-d126-4ed4-b5da-cf63df3d65b0.png)
![Adding a new Permission](https://user-images.githubusercontent.com/81187224/174974966-37125900-ee63-4491-86cb-f8a50360a34f.png)
![Adding Permission to Role](https://user-images.githubusercontent.com/81187224/174975052-6d64cd2d-117a-4e89-b8b7-8863a902d31e.png)
